packages/test-generator/integration-test-templates/src/FormTests/DSBidirectionalToy.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+import '@aws-amplify/ui-react/styles.css';
+import { AmplifyProvider, View, Heading, Text } from '@aws-amplify/ui-react';
+import { useState, useEffect, useRef } from 'react';
+import { DataStore } from '@aws-amplify/datastore';
+import { DataStoreFormCreateBidirectionalDog } from '../ui-components'; // eslint-disable-line import/extensions
+import { BiDirectionalDog, BiDirectionalToy } from '../models';
+
+const toyName = 'Bone';
+
+const initializeTestData = async (): Promise<{
+  connectedDog: BiDirectionalDog;
+}> => {
+  const connectedDog = await DataStore.save(new BiDirectionalDog({ name: 'Fluffy' }));
+  const connectedToy = await DataStore.save(
+    new BiDirectionalToy({
+      name: toyName,
+      BiDirectionalDog: connectedDog,
+      biDirectionalDogID: connectedDog.id,
+      biDirectionalDogBiDirectionalToysId: connectedDog.id,
+    }),
+  );
+
+  await DataStore.save(
+    BiDirectionalDog.copyOf(connectedDog, (updated) => {
+      Object.assign(updated, {
+        BiDirectionalToy: connectedToy,
+      });
+    }),
+  );
```

Comment:
Connecting Toy to Dog should happen automatically when BiDirectionalToy is saved. Reference [DSBidirectionalDog.tsx](https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/test-generator/integration-test-templates/src/FormTests/DSBidirectionalDog.tsx)